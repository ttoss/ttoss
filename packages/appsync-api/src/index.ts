export { type ResolverResolveParams, schemaComposer } from 'graphql-compose';
export { createApiTemplate } from './createApiTemplate';
export {
  type AppSyncResolverHandler,
  createAppSyncResolverHandler,
} from './createAppSyncResolverHandler';
