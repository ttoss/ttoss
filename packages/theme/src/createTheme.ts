import { BruttalTheme } from './themes/Bruttal/Bruttal';
import deepmerge from 'deepmerge';
import type { Theme } from 'theme-ui';

export const createTheme = (theme: Theme): Theme => {
  return deepmerge(BruttalTheme, theme);
};
