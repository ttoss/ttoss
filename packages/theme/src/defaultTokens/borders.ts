/**
 * https://github.com/chakra-ui/chakra-ui/blob/4eca0a6bb5f3db8e4edc0b875e0f22f4c46519d9/packages/react/src/theme/tokens/borders.ts#L4
 */
export const borders = Object.freeze({
  none: '0px solid',
  xs: '0.5px solid',
  sm: '1px solid',
  md: '2px solid',
  lg: '4px solid',
  xl: '8px solid',
});
