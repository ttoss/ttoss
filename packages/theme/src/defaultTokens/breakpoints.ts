/**
 * https://www.chakra-ui.com/docs/theming/breakpoints
 */
export const breakpoints = Object.freeze({
  sm: '640px', // 40em
  md: '768px', // 48em
  lg: '1024px', // 64em
  xl: '1280px', // 80em
  '2xl': '1536px', // 96em
});
