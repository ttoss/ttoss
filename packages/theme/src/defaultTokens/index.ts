export { radii } from './borderRadius';
export { borders } from './borders';
export { breakpoints } from './breakpoints';
export { sizes } from './sizes';
export { spacing } from './spacing';
export {
  fontSizes,
  fontWeights,
  letterSpacings,
  lineHeights,
} from './typography';
export { zIndices } from './zIndex';
