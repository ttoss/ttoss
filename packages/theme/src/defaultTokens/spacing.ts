/**
 * https://www.chakra-ui.com/docs/theming/spacing
 */
export const spacing = Object.freeze({
  px: '1px', // 1px
  0.5: '0.125rem', // 2px
  1: '0.25rem', // 4px
  1.5: '0.375rem', // 6px
  2: '0.5rem', // 8px
  2.5: '0.625rem', // 10px
  3: '0.75rem', // 12px
  3.5: '0.875rem', // 14px
  4: '1rem', // 16px
  5: '1.25rem', // 20px
  6: '1.5rem', // 24px
  7: '1.75rem', // 28px
  8: '2rem', // 32px
  9: '2.25rem', // 36px
  10: '2.5rem', // 40px
  12: '3rem', // 48px
  14: '3.5rem', // 56px
  16: '4rem', // 64px
  20: '5rem', // 80px
  24: '6rem', // 96px
  28: '7rem',
  32: '8rem',
  36: '9rem',
  40: '10rem',
  44: '11rem',
  48: '12rem',
  52: '13rem',
  56: '14rem',
  60: '15rem',
  64: '16rem',
  72: '18rem',
  80: '20rem',
  96: '24rem',
});
