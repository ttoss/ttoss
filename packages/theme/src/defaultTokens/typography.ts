/**
 * https://www.chakra-ui.com/docs/theming/typography#font-sizes
 */
export const fontSizes = Object.freeze({
  '4xs': '0.375rem', // 6px
  '3xs': '0.5rem', // 8px
  '2xs': '0.625rem', // 10px
  xs: '0.75rem', // 12px
  sm: '0.875rem', // 14px
  md: '1rem', // 16px
  lg: '1.125rem', // 18px
  xl: '1.25rem', // 20px
  '2xl': '1.5rem', // 24px
  '3xl': '1.875rem', // 30px
  '4xl': '2.25rem', // 36px
  '5xl': '3rem', // 48px
  '6xl': '3.75rem', // 60px
  '7xl': '4.5rem', // 72px
  '8xl': '6rem', // 96px
  '9xl': '8rem', // 128px
});

/**
 * https://www.chakra-ui.com/docs/theming/typography#font-weights
 */
export const fontWeights = Object.freeze({
  hairline: 100,
  thin: 200,
  light: 300,
  normal: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900,
});

/**
 * https://www.chakra-ui.com/docs/theming/typography#line-heights
 */
export const lineHeights = Object.freeze({
  normal: 'normal',
  none: 1,
  shorter: 1.25,
  short: 1.375,
  moderate: 1.5,
  tall: 1.625,
  taller: '2',
});

/**
 * https://www.chakra-ui.com/docs/theming/typography#letter-spacings
 */
export const letterSpacings = Object.freeze({
  tighter: '-0.05em',
  tight: '-0.025em',
  normal: '0',
  wide: '0.025em',
  wider: '0.05em',
  widest: '0.1em',
});
