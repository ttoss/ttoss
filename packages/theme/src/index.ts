export { createTheme } from './createTheme';
export { addIcon } from '@iconify-icon/react';
