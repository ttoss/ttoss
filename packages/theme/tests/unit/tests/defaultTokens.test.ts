import * as defaultTokens from 'src/defaultTokens';

test('defaultTokens cannot be changed', () => {
  expect(defaultTokens).toMatchSnapshot();
});
