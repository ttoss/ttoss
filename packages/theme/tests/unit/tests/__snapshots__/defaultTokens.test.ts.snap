// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultTokens cannot be changed 1`] = `
{
  "__esModule": true,
  "breakpoints": {
    "2xl": "1536px",
    "lg": "1024px",
    "md": "768px",
    "sm": "640px",
    "xl": "1280px",
  },
  "fontSizes": {
    "2xl": "1.5rem",
    "2xs": "0.625rem",
    "3xl": "1.875rem",
    "3xs": "0.5rem",
    "4xl": "2.25rem",
    "4xs": "0.375rem",
    "5xl": "3rem",
    "6xl": "3.75rem",
    "7xl": "4.5rem",
    "8xl": "6rem",
    "9xl": "8rem",
    "lg": "1.125rem",
    "md": "1rem",
    "sm": "0.875rem",
    "xl": "1.25rem",
    "xs": "0.75rem",
  },
  "fontWeights": {
    "black": 900,
    "bold": 700,
    "extrabold": 800,
    "hairline": 100,
    "light": 300,
    "medium": 500,
    "normal": 400,
    "semibold": 600,
    "thin": 200,
  },
  "letterSpacings": {
    "normal": "0",
    "tight": "-0.025em",
    "tighter": "-0.05em",
    "wide": "0.025em",
    "wider": "0.05em",
    "widest": "0.1em",
  },
  "lineHeights": {
    "moderate": 1.5,
    "none": 1,
    "normal": "normal",
    "short": 1.375,
    "shorter": 1.25,
    "tall": 1.625,
    "taller": "2",
  },
  "radii": {
    "2xl": "1rem",
    "2xs": "0.0625rem",
    "3xl": "1.5rem",
    "4xl": "2rem",
    "full": "9999px",
    "lg": "0.5rem",
    "md": "0.375rem",
    "none": "0",
    "sm": "0.25rem",
    "xl": "0.75rem",
    "xs": "0.125rem",
  },
  "sizes": {
    "2xl": "42rem",
    "2xs": "16rem",
    "3xl": "48rem",
    "3xs": "14rem",
    "4xl": "56rem",
    "5xl": "64rem",
    "6xl": "72rem",
    "7xl": "80rem",
    "8xl": "90rem",
    "container": {
      "lg": "1024px",
      "md": "768px",
      "sm": "640px",
      "xl": "1280px",
    },
    "full": "100%",
    "lg": "32rem",
    "max": "max-content",
    "md": "28rem",
    "min": "min-content",
    "sm": "24rem",
    "xl": "36rem",
    "xs": "20rem",
  },
  "spacing": {
    "0.5": "0.125rem",
    "1": "0.25rem",
    "1.5": "0.375rem",
    "10": "2.5rem",
    "12": "3rem",
    "14": "3.5rem",
    "16": "4rem",
    "2": "0.5rem",
    "2.5": "0.625rem",
    "20": "5rem",
    "24": "6rem",
    "28": "7rem",
    "3": "0.75rem",
    "3.5": "0.875rem",
    "32": "8rem",
    "36": "9rem",
    "4": "1rem",
    "40": "10rem",
    "44": "11rem",
    "48": "12rem",
    "5": "1.25rem",
    "52": "13rem",
    "56": "14rem",
    "6": "1.5rem",
    "60": "15rem",
    "64": "16rem",
    "7": "1.75rem",
    "72": "18rem",
    "8": "2rem",
    "80": "20rem",
    "9": "2.25rem",
    "96": "24rem",
    "px": "1px",
  },
  "zIndices": {
    "auto": "auto",
    "banner": 1200,
    "base": 0,
    "docked": 10,
    "dropdown": 1000,
    "hide": -1,
    "max": 2147483647,
    "modal": 1400,
    "overlay": 1300,
    "popover": 1500,
    "skipNav": 1600,
    "sticky": 1100,
    "toast": 1700,
    "tooltip": 1800,
  },
}
`;
